pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go

```diff
@@ -82,6 +82,13 @@ var _ = Describe("Test ClusterResourcePlacement Watcher", func() {
 
 	Context("When updating clusterResourcePlacement", func() {
 		BeforeEach(func() {
+			By("By checking the placement queue before resetting")
+			// The event could arrive after the resetting, which causes the flakiness.
+			// It makes sure the queue is clear before proceed.
+			Eventually(func() bool {
+				return fakePlacementController.Key() == testCRPName
+			}, eventuallyTimeout, interval).Should(BeTrue(), "placementController should receive the CRP name")
```

Comment:
I don't fully get it. If we want to get a clean start for all the test, why create the CPR in the first place at line 60 and then reset it before each test?

Reply:
one can move the create to the test in line 75

Reply:
and here

Reply:
or move the test in line 75 to the BeforeEach for all.

Reply:
moved the create test into the BeforeEach.